test/integration/warm-pool/test_fns.go

```diff
@@ -0,0 +1,102 @@
+package warm_pool
+
+import (
+	"fmt"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	"strconv"
+)
+
+// Scales the cluster from minPods up to maxPods and back down to minPods
+func QuickScale(f *framework.Framework, deploymentSpec *manifest.DeploymentBuilder) (*v1.Deployment, error) {
+	// create deployment
+	deployment := deploymentSpec.Build()
+
+	_, err := f.K8sResourceManagers.
+		DeploymentManager().
+		CreateAndWaitTillDeploymentIsReady(deployment, utils.DefaultDeploymentReadyTimeout*5)
+	Expect(err).ToNot(HaveOccurred())
+
+	fmt.Fprintf(GinkgoWriter, "Scaling cluster up to %v pods\n", maxPods)
+	changeReplicas(deploymentSpec, maxPods)
+
+	Expect(maxPods).To(Equal(busyboxPodCnt()))
```

Comment:
Instead of using `Expect` in these functions, they should return an `error` object so that the caller can catch it